geopmdpy/geopmdpy/session.py

```diff
@@ -185,33 +282,37 @@ def run(self, run_time, period, pid, print_header,
                                  CSV data.
 
             request_stream (typing.IO): Input from user describing the
-                                   requests to read.
+                                        requests to read.
 
             out_stream (typing.IO): Stream where output from will be
-                               printed.
+                                    printed.
 
         """
-        self._requests = ReadRequestQueue(request_stream)
-        do_stats = report_stream is not None
+        if session_io is not None:
+            if request_stream is not None:
+                raise RuntimeError('Using a request_stream is incompatible with using session_io')
+            if report_path is not None:
+                raise RuntimeError('Using a report_path is incompatible with using session_io')
+            requests = session_io.get_request_queue()
+            do_stats = session_io.do_report()
+        else:
+            requests = ReadRequestQueue(request_stream)
+            do_stats = report_path is not None
         self.check_read_args(run_time, period)
+        signal_config = [(rr[0], rr[1], rr[2]) for rr in requests]
```

Comment:
```suggestion
        signal_config = list(requests)
```
(if the goal is just to convert the iterable to a list)